spec/spec_helper.rb

```diff
@@ -1 +1,23 @@
+if ENV['COVERAGE']
+  # In order to archive an expected coverage analysis we need to eager load
+  # all logstash code base, otherwise it will not get a good analysis.
+
+  require "pluginmanager/main"
+  Dir.glob("lib/logstash/patches/*.rb") do |file|
+    require file
+  end
+
+  Dir.glob("lib/logstash/*.rb") do |file|
+    require file
+  end
+
+  Dir.glob("lib/logstash/util/*.rb") do |file|
+    next if ["lib/logstash/util/prctl.rb"].include?(file)
+    require file
```

Comment:
makes sense, I will wrap this is a more decent method + exclude_list before merging this in. Thanks for your time reviewing @colinsurprenant